T22: Setup prisma and users table

hferoz1 opened the pull request:
Setup prisma ORM to interact with the database and create a User model.

- downloaded prisma (prisma.schema file)
- created a db.js file 
- created a server.js file
- added .env file to .gitignore
- created a basic User model 

I just wanted to make sure I created the ORM properly and am using it as intended. I am still a little iffy on ORMS's and prisma so detailed feedback would be great!

hferoz1 committed:
T22: Setup prisma and users table

Setup prisma ORM to interact with the database and create a User model.

- downloaded prisma (prisma.schema file)
- created a db.js file 
- created a server.js file
- added .env file to .gitignore
- created a basic User model 

I just wanted to make sure I created the ORM properly and am using it as intended. I am still a little iffy on ORMS's and prisma so detailed feedback would be great!

fahadakhan2 commented on server/prisma/dev.db:
This file and the dev.db-journal should not be committed into github just like the .env. Go ahead and put these file names inside your gitignore as well

fahadakhan2 commented on server/prisma/migrations/20240701151138_init/migration.sql:
```diff
@@ -0,0 +1,9 @@
+-- CreateTable
+CREATE TABLE "Post" (
```
Looks like what happened here was that originally in schema.prisma you had the model named as Post, then ran a migration. And then later you went and changed the model in the schema.prisma file to User. Since you changed your schema again you should delete this migration folder and run a new migration to reflect your changes.

fahadakhan2 commented on server/prisma/schema.prisma:
```diff
@@ -0,0 +1,18 @@
+}
+
+model User {
+  id       Int      @id @default(uuid())
```
id should be String instead of Int

fahadakhan2 commented on server/prisma/schema.prisma:
```diff
@@ -0,0 +1,18 @@
+  id       Int      @id @default(uuid())
+  email    String   @unique
+  username String?
+  name     String
```
I'm guessing this represent full name so to be more descriptive change the field name to fullName.

fahadakhan2 commented on server/prisma/schema.prisma:
```diff
@@ -0,0 +1,18 @@
+  url      = env("DATABASE_URL")
+}
+
+model User {
```
A barebone user table usually has additional fields alongside what you have currently such as:

  password  (stores the password of the user)
  createdAt (when the users account was created)
  updatedAt (if the user updates there profile for example this is a timestamp to show when they updated their information)

fahadakhan2 commented on server/server/server.js:
```diff
@@ -0,0 +1,27 @@
+const prisma = require('./db/db')
+const express = require('express');
+const { PrismaClient } = require('@prisma/client');
+const prisma = new PrismaClient();
```
Not needed. You have this same logic extracted in the db/db.js folder so when you want to use it you should import from there

fahadakhan2 commented on server/server/server.js:
```diff
@@ -0,0 +1,27 @@
+const prisma = require('./db/db')
```
This task is only to set up prisma and the users table so there shouldn't need to be any additions of code in the server.js file

fahadakhan2 commented on server/server/server.js:
```diff
@@ -0,0 +1,27 @@
+
+app.use(express.json());
+
+app.post('/users', async (req, res) => {
```
this logic is for a different task on airtable which is the sign up api routes. Remove this logic from this PR but if you'd like make another PR for the other task since you basically have most the logic here you can go ahead and do that.

fahadakhan2 reviewed (changes requested):
Good start! Needs some changes and then we can get it merged.

hferoz1 committed:
Deleted server directory to add updated files

hferoz1 committed:
Updated PR -- Changes Made

- removed the dev.db, dev.db-journal. and .env files from the pull request and added them to the .gitignore
- created a new migration file that aligned with my schema.prisma
- fixed the data types (ex. id data type is now String instead of int)
- added more information to the schema.prisma file (password, createdAt, firstName, etc)
- turned the extra code in my server.js file (app.post('/users', async (req, res)) into comments for now

hferoz1 committed:
Delete frontend/database directory

hferoz1 committed:
added the database under the backend directory

Accidentally added it under frontend before so I had to delete it and put it under backend

fahadakhan2 reviewed (approved):
lgtm you can go ahead and merge!
